lib/module.js

```diff
@@ -121,6 +122,10 @@ function axiosModule (_moduleOptions) {
   // Set _AXIOS_BASE_URL_ for dynamic SSR baseURL
   process.env._AXIOS_BASE_URL_ = options.baseURL
 
+  // Create a separate instance for server middlewares.
+  // Extended from a server-side plugin.
+  process.axios = Axios.create({ baseURL: options.baseURL })
```

Comment:
We can attach axios instance to `nuxt` (for other modules) and `req` object (for server middleware) to avoid leaking references to global

Reply:
To attach to `req` we would only be able to do it when a request is actually made. This here creates `process.axios` earlier on purpose so that there is no point during runtime where server middleware would see undefined `process.axios`.

Reply:
Hmm, I think the current solution has a problem that in the first request to server-middleware the axios instance would not be extended yet (since the server-side plugin would run after server middleware).

So that pertains to the second issue you've made below -- we would need to probably extract the "extending" code into a separate file and call it immediately rather than relying on plugin extending it.

Reply:
And then putting in on `req` would actually be possible from the module and potentially be a better idea. It would mean that the instance of axios would be a bit harder to access (would have to be passed along various calls potentially) but workable.